esx_service/utils/vmdk_utils.py

```diff
@@ -274,26 +277,25 @@ def get_vm_uuid_by_name(vm_name):
     """ Returns vm_uuid for given vm_name, or None """
     si = vmdk_ops.get_si()
     try:
-        vm = [d for d in si.content.rootFolder.childEntity[0].vmFolder.childEntity if d.config.name == vm_name]
-        return vm[0].config.uuid
+        vm = FindChild(GetVmFolder(), vm_name)
+        return vm.config.uuid
     except:
         return None
 
 def get_vm_name_by_uuid(vm_uuid):
     """ Returns vm_name for given vm_uuid, or None """
     si = vmdk_ops.get_si()
     try:
-        vm = [d for d in si.content.rootFolder.childEntity[0].vmFolder.childEntity if d.config.uuid == vm_uuid]
-        return vm[0].config.name
+        return vmdk_ops.vm_uuid2name(vm_uuid)
```

Comment:
Good catch! Yes, I noticed that too. Rather code in vmdk_ops.py triggered light bulbs! I wanted to separate bug fix from code cleanup so didnt touch vm_uuid2name